client.go

```diff
@@ -1524,12 +1576,235 @@ func (cl *Client) AddTorrentInfoHash(infoHash metainfo.Hash) (t *Torrent, new bo
+	CompletedDate int64 `json:"completed_date,omitempty"`
+
+	// time
+	DownloadingTime int64 `json:"downloading_time,omitempty"`
+	SeedingTime     int64 `json:"seeding_time,omitempty"`
+
+	// .torrent
+	Comment   string `json:"comment,omitempty"`
+	Creator   string `json:"creator,omitempty"`
+	CreatedOn int64  `json:"created_on,omitempty"`
+}
+
+// Save torrent to state file
+func (cl *Client) SaveTorrent(t *Torrent) ([]byte, error) {
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+
+	s := TorrentState{Version: 1}
+
+	if t.info != nil {
+		s.MetaInfo = t.newMetaInfo()
+	} else {
+		s.InfoHash = &t.infoHash
+		s.Name = t.name()
+		s.Trackers = t.announceList()
+	}
+
+	if cl.activeTorrent(t) {
+		now := time.Now().Unix()
+		if t.seeding() {
+			t.seedingTime = t.seedingTime + (now - t.activateDate)
+		} else {
+			t.downloadingTime = t.downloadingTime + (now - t.activateDate)
+		}
+		t.activateDate = now
+	}
+
+	s.Downloaded = t.downloaded
+	s.Uploaded = t.uploaded
+
+	s.Checks = t.checks
+
+	s.DownloadingTime = t.downloadingTime
+	s.SeedingTime = t.seedingTime
+
+	s.AddedDate = t.addedDate
+	s.CompletedDate = t.completedDate
+
+	s.Comment = t.comment
+	s.Creator = t.creator
+	s.CreatedOn = t.createdOn
+
+	if t.info != nil {
+		s.Pieces = t.bitfield()
+	}
+
+	return json.Marshal(s)
+}
+
+// Load torrent from saved state
+func (cl *Client) LoadTorrent(buf []byte) (t *Torrent, err error) {
+	var s TorrentState
+	json.Unmarshal(buf, &s)
+
+	var spec *TorrentSpec
+
+	if s.MetaInfo == nil {
+		spec = &TorrentSpec{
+			Trackers:    s.Trackers,
+			DisplayName: s.Name,
+			InfoHash:    *s.InfoHash,
+		}
+	} else {
+		spec = TorrentSpecFromMetaInfo(s.MetaInfo)
+	}
+
+	var n bool
+	t, n = cl.AddTorrentInfoHash(spec.InfoHash)
+	if !n {
+		err = errors.New("already exists")
+		t = nil
+		return
+	}
+	if spec.DisplayName != "" {
+		t.SetDisplayName(spec.DisplayName)
+	}
+	if spec.Info != nil {
+		if s.Pieces == nil {
+			s.Pieces = make([]bool, spec.Info.NumPieces())
+		}
+		err = t.loadTorrent(spec.Info.Bytes, s.Pieces, s.Checks)
+		if err != nil {
+			return
+		}
+	}
+
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+
+	if t.info != nil {
+		t.fileUpdateCheck()
+	}
+
+	if spec.ChunkSize != 0 {
+		t.chunkSize = pp.Integer(spec.ChunkSize)
+	}
+	t.addTrackers(spec.Trackers)
+
+	t.downloaded = s.Downloaded
+	t.uploaded = s.Uploaded
+
+	t.downloadingTime = s.DownloadingTime
+	t.seedingTime = s.SeedingTime
+
+	t.addedDate = s.AddedDate
+	t.completedDate = s.CompletedDate
+
+	t.comment = s.Comment
+	t.creator = s.Creator
+	t.createdOn = s.CreatedOn
+
+	return
+}
+
+// Do any network actvity only after calling this function. We can add torrent,
+// run file consictency checks and then manually start it. Here are two
+// possibilites: 1) Create *Torrent without adding it to client. 2) Add Torrent
+// to client and do not announce it or download.
+func (cl *Client) StartTorrent(t *Torrent) error {
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+
+	return cl.startTorrent(t)
+}
+
```

Comment:
No downloading occurs other than to fetch metainfo until the Torrent is instructed which pieces are wanted. It seems that you want network activity, even just for the metainfo, even in your case?